### PR TITLE
post added

### LINKED ISSUE #3: 2. POST course
**As a site visitor, I want to be able to post new object from postman and render it on the terminal or localhost with the other lists.**

This user story has been developed through a branch called 'post'.

> Assign to **Gelila Ali*

### Node.js

- import express framework
- import fs module
- import joi  framework
- write logic to post a new list on the postman

### PR DESCRIPTION
**Pull request by gelilaa**
This closes #3

### CODE REVIEW

**Comment by aleks2407** [comment]
@gelilaa I'm a bit lost for now, so can't really give a good feedback, i guess))) Mamé will be probably a better person to review the PR :)

**Review by mametur — APPROVED** [review]
Works fine for me :+1: :

**Comment by mametur** [comment]
@gelilaa I have forgotten adding an error handling  to fs.readmeFile and you have forgotten adding it to fs.writemefile as well:) if we can add them I guess it will be better.

**Comment by gelilaa** [comment]
> @gelilaa I have forgotten adding an error handling to fs.readmeFile and you have forgotten adding it to fs.writemefile as well:) if we can add them I guess it will be better.


 Do you mean `try` and `catch`?

**Comment by mametur** [comment]
> > @gelilaa I have forgotten adding an error handling to fs.readmeFile and you have forgotten adding it to fs.writemefile as well:) if we can add them I guess it will be better.
> 
> Do you mean `try` and `catch`?

fs.readFile('/foo.txt', function(err, data) {
  // If an error occurred, handle it (throw, propagate, etc)
  if(err) {
    console.log('Unknown Error');
    return;
  }
  // Otherwise, log the file contents
  console.log(data);
});

**Comment by mametur** [comment]
@gelilaa  https://fredkschott.com/post/2014/03/understanding-error-first-callbacks-in-node-js/

**Comment by gelilaa** [comment]
> > > @gelilaa I have forgotten adding an error handling to fs.readmeFile and you have forgotten adding it to fs.writemefile as well:) if we can add them I guess it will be better.
> > 
> > 
> > Do you mean `try` and `catch`?
> 
> fs.readFile('/foo.txt', function(err, data) {
> // If an error occurred, handle it (throw, propagate, etc)
> if(err) {
> console.log('Unknown Error');
> return;
> }
> // Otherwise, log the file contents
> console.log(data);
> });


Got it!

**Review by aleks2407 — APPROVED** [review]
Good job, thank you @gelilaa ;)

**Comment by gelilaa** [comment]
@mametur As Agreed I have added callbacks for `fs.readfile()` and `fs.writeFile()`. check it out and let me know what you think .

File: index.js [hunk 1]
```diff
@@ -13,16 +13,24 @@ app.get('/', function (req, res) {
...
-const parsedContent = JSON.parse(content);
+
+// callback is added
+const content = fs.readFile(filePath, 'utf-8',(err,data)=>{
+	if(err){
+		console.error(err)
+	}
+	JSON.parse(data)
+});
+
```
Comment (mametur) [inline index.js]: This function should be in (get and post  methods) otherwise you can not access fs.readFile data  because they work async .

**Pull request merged**